src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -377,6 +378,66 @@ def list_key(cmd,
         raise CLIError(str(exception))
 
 
+def restore_key(cmd,
+                datetime,
+                key=None,
+                name=None,
+                label=None,
+                no_deletes=False,
+                connection_string=None,
+                yes=False):
+    current_retry = 0
+    max_retry = 3
+    retry_interval = 1
+
+    connection_string = resolve_connection_string(cmd, name, connection_string)
+    azconfig_client = AzconfigClient(connection_string)
+
+    query_option_then = QueryKeyValueCollectionOptions(key_filter=key,
+                                                       label_filter=QueryKeyValueCollectionOptions.empty_label if label is not None and not label else label,
+                                                       query_datetime=datetime)
+    query_option_now = QueryKeyValueCollectionOptions(key_filter=key,
+                                                      label_filter=QueryKeyValueCollectionOptions.empty_label if label is not None and not label else label)
+
+    try:
+        restore_keyvalues = azconfig_client.get_keyvalues(query_option_then)
+        current_keyvalues = azconfig_client.get_keyvalues(query_option_now)
```

Comment:
This can be simplified by calling list(). No need to create QueryKeyValueCollectionOptions
key_values = list_key(cmd,
                                  key=key,
                                  label=label,
                                  name=name,
                                  datetime=datetime,
                                  connection_string=connection_string,
                                  all_=True)

Reply:
After retrieving kvs, list_key loops through the keyvalue_iterable, putting them into another list. All I need is the iterable here. If the below table is correct, we can use that as a comment.

If label is            |       label_filter
None                        None
'' (empty string)        empty_label
anything else            label

Or simply:
if (not label)
     label = None

Reply:
In your case, I think if label == None, it means all labels. If label is empty string (--label ""), then it means null label. As in general, null label == empty label. While, for different commands, if client doesn't specify --label, it may have different meanings.

Reply:
Oh, you're right. My comment with if (not label) is wrong. Maybe this would be clearer:

if label==''
    label = QueryKeyValueCollectionOptions.empty_label

then in the QueryKeyValueCollectionOptions:

label_filter = label

Because, really the confusing part is the ternary in the label_filter. The implementation for restore is identical to that of list. The help reflects that, too.

Reply:
That looks good.

Reply:
I made the change here to simplify, but left the other locations alone where that ternary logic is used. If we feel like we should simplify those, too, we should file a different issue.